music/app/controllers/services_controller.rb

```diff
@@ -11,7 +11,9 @@ def new
   def create
     @resource = OpenStruct.new(date: Date.parse(params[:resource][:date]), body: params[:resource][:body])
```

Comment:
So if creating the object fails, the `new` template is re-rendered using this `@resource` to prefill the form. Normally this would be done with a model object but because this form doesn't map straight to a model, I'm faking it with OpenStruct. The provided `:congregation_id` should really be included in this command and you should check that the form remembers the value if it errors.

PS. Was going to suggest you add a test for that but... I have no testing framework for this project. So I'll take your word for it.

PPS. If for some unfathomable reason you want to add tests, use rspec, not the builtin test framework